test/core/store/fs/test_registry.py

```diff
@@ -137,34 +138,61 @@ def test_mldataset_levels(self):
 
     def test_dataset_zarr(self):
         data_store = self.create_data_store()
-        self._assert_dataset_format_supported(data_store, '.zarr')
+        self._assert_dataset_supported(
+            data_store,
+            filename_ext='.zarr',
+            requested_dtype_alias=None,
+            expected_dtype_aliases={'dataset'},
+            expected_return_type=xr.Dataset,
+            expected_descriptor_type=DatasetDescriptor
+        )
 
     def test_dataset_netcdf(self):
         data_store = self.create_data_store()
-        self._assert_dataset_format_supported(data_store, '.nc')
+        self._assert_dataset_supported(
+            data_store,
+            filename_ext='.nc',
+            requested_dtype_alias=None,
+            expected_dtype_aliases={'dataset'},
+            expected_return_type=xr.Dataset,
+            expected_descriptor_type=DatasetDescriptor
+        )
+
+    def test_dataset_levels(self):
+        data_store = self.create_data_store()
+        self._assert_dataset_supported(
+            data_store,
+            filename_ext='.levels',
+            requested_dtype_alias='dataset',
+            expected_dtype_aliases={'mldataset', 'dataset'},
+            expected_return_type=xr.Dataset,
+            expected_descriptor_type=None
```

Comment:
So, we can use the store to open a dataset, but not to get a descriptor? This does not seem right.

Reply:
`expected_descriptor_type=None` means we do not want to test for the descriptor, because test code would be too confusing. It does not mean, the descriptor must not exist.

Reply:
I'm sorry, but to me it looks like we are configuring the tests in a way that they run green. It should work if we specify expected_descriptor_type=DatasetDescriptor , just like in the other tests.

Reply:
>> It should work if we specify expected_descriptor_type=DatasetDescriptor , just like in the other tests.

No, because the descriptor is retrieved using `search_data(expected_data_type)`. But this will not return a descriptor for dataset (yet), because it is currently implemented to return MultiLevelDatasetDescriptorsa only.

Again, this is not part of this PR.

Reply:
Okay, then please go ahead and merge.